checkov/main.py

```diff
@@ -144,7 +146,11 @@ def run(banner: str = checkov_banner, argv: list[str] = sys.argv[1:]) -> int | N
         "python_executable": sys.executable,
         "python_version": sys.version,
         "checkov_executable": sys.argv[0],
-        "args": parser.format_values(sanitize=True).split('\n')
+        "args": parser.format_values(sanitize=True).split('\n'),
+        "OS_system_info": platform.platform(),
+        "CPU_architecture": platform.processor(),
+        "Python_implementation": platform.python_implementation()
+
     }
 
     logger.debug(f'Run metadata: {json.dumps(run_metadata, indent=2)}')
```

Comment:
Can we add some logic somewhere around here to ensure the following:

- If --support is set, then check for a non-empty API key and log a warning if it is not set: "The --support flag requires a platform API key in order to upload the logs."
- If --support is set, set the log level to DEBUG (the logger is already initialized, so you'll have to update it)

Reply:
We need to set debug logs at least before we print the run metadata, because that is the first piece of critical information

Reply:
@mikeurbanski1 Just a quick thought regarding the 2nd one, can it be something else and not only DEBUG?

Reply:
Do you mean if the user runs `checkov --support` a log warning should appear in his console?

Reply:
as far as I can tell you already implemented the first part, by logging an error, if someone uses `--support` flag without API key, but please move it to the beginning of the actual run method not in the `finally` block.
for the second one, you need to adjust the log level of the `stream_handler` to `DEBUG` this should be possible by just adding
```python
stream_handler.setLevel(logging.DEBUG)
```

Reply:
The second point, I have already added it. 
The first one, if I put it at the beginning of the run method. It will send the stream string to our file in s3 and it will not contains all the logs. And if I put it at the end of the run method without (try and finally) and the user has an error or expiation the run will stop when rising the bug to the console and the lines of handling the support flag will not be run or save. So, I used `try and finally` to run the support flag with all logs even if there are exceptions they will be saved.